File: Python/Deber 01/calculadora.py
operacion = input("Seleccione operacion\n")

if (operacion == "suma"):
    numero1 = int(input(" Ingrese numero 1\n"))
    numero2 = int(input(" Ingrese numero 2\n"))
    print ("La suma es:", numero1 + numero2)
else:
    if (operacion == "resta"):
        numero1 = int(input(" Ingrese numero 1\n"))
        numero2 = int(input(" Ingrese numero 2\n"))
        print("La resta es:", numero1 - numero2)
    else:
        if (operacion == "multiplicacion"):
            numero1 = int(input(" Ingrese numero 1\n"))
            numero2 = int(input(" Ingrese numero 2\n"))
            print("La multiplicacion es:", numero1 * numero2)
        else:
            if (operacion == "division"):
                numero1 = int(input(" Ingrese numero 1\n"))
                numero2 = int(input(" Ingrese numero 2\n"))
                print("La division es:", numero1 / numero2)
            else:
                if ( operacion == "pendiente de una recta"):
                    puntoY2 = int(input(" Ingrese punto Y2\n"))
                    puntoY1 = int(input(" Ingrese punto Y1\n"))
                    puntoX2 = int(input(" Ingrese punto X2\n"))
                    puntoX1 = int(input(" Ingrese punto X1\n"))
                    print( "La pendiente de la recta es:", ( puntoY2 - puntoY1 ) / ( puntoX2 - puntoX1 ))
                else:
                    print("No es una operacion valida");

File: 01-Introduccion/calculadora.ts
const operaciones = prompt(" Operacion a realizar ");
const operacion1 = "suma"
const operacion2 = "resta"
const operacion3 = "multiplicacion"
const operacion4 = "division"
const operacion5 = "pendiente de una recta"
const operacion6 = "distancia de dos puntos"



if(operaciones == "suma") {
    const numerounoString = prompt("Ingrese primer numero");
    const numerodosString = prompt("Ingrese segundo numero");
    const numero1 = Number(numerounoString); 
    const numero2 = Number(numerodosString); 
    const suma = numero1 + numero2;
    console.log("El resultado es: " ,suma);
    } else { 
        if(operaciones == operacion2){
        const numerounoString = prompt("Ingrese primer numero");
        const numerodosString = prompt("Ingrese segundo numero");
        const numero1 = Number(numerounoString); 
        const numero2 = Number(numerodosString); 
        const resta = numero1 - numero2;
        console.log("El resultado es: " ,resta);
            }else{
                if(operaciones == operacion4){
                const numerounoString = prompt("Ingrese primer numero");
                const numerodosString = prompt("Ingrese segundo numero");
                const numero1 = Number(numerounoString); 
                const numero2 = Number(numerodosString); 
                const division = numero1 / numero2;
                console.log("El resultado es: " ,division);    
                }else{
                     if(operaciones == operacion3){
                    const numerounoString = prompt("Ingrese primer numero");
                    const numerodosString = prompt("Ingrese segundo numero");
                    const numero1 = Number(numerounoString); 
                    const numero2 = Number(numerodosString); 
                    const multiplicacion = numero1 * numero2;
                    console.log("El resultado es: " ,multiplicacion);
                    }else{
                        if(operaciones == operacion5){
                            const numerotresString = prompt("Ingrese puntoY2");
                            const numerocuatroString = prompt("Ingrese puntoY1");
                            const numerocincoString = prompt("Ingrese puntoX2");
                            const numeroseisString = prompt("Ingrese puntoX1");
                            const puntoY2 = Number(numerotresString); 
                            const puntoY1 = Number(numerocuatroString); 
                            const puntoX2 = Number(numerocincoString); 
                            const puntoX1 = Number(numeroseisString); 
                            const pendienteDeUnaRecta = (puntoY2 - puntoY1)/ (puntoX2 - puntoX1);
                            console.log("El resultado es: ", pendienteDeUnaRecta);
                        }else{
                            if(operaciones == operacion6){
                                const numerotresString = prompt("Ingrese puntoY2");
                                const numerocuatroString = prompt("Ingrese puntoY1");
                                const numerocincoString = prompt("Ingrese puntoX2");
                                const numeroseisString = prompt("Ingrese puntoX1");
                                const puntoY2 = Number(numerotresString); 
                                const puntoY1 = Number(numerocuatroString); 
                                const puntoX2 = Number(numerocincoString); 
                                const puntoX1 = Number(numeroseisString); 
                                const distanciaDeDosPuntos = (((puntoY2 - puntoY1)^2) + ((puntoX2 - puntoX1))^2)^0.5;
                                console.log("El resultado es: ", distanciaDeDosPuntos );
                                }else{
                                    console.log("La operacon es invalida");
                            }
                         }
                     } 
                } 
            }
        }   